### PR TITLE
rustup https://github.com/rust-lang/rust/pull/71292/

### PR DESCRIPTION
**Pull request by matthiaskrgr**
cc https://github.com/rust-lang/rust/issues/71608

---

changelog: none

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
Something with caching on CI is odd https://github.com/rust-lang/rust-clippy/pull/5543/checks?check_run_id=626103141#step:31:1

**Comment by flip1995** [comment]
All the calls to `expect_local` make me a bit nervous. But if they should panic, this is hopefully caught by our integration tests.

@bors r+

File: clippy_lints/src/functions.rs [hunk 1]
```diff
@@ -489,7 +489,12 @@ fn is_mutable_pat(cx: &LateContext<'_, '_>, pat: &hir::Pat<'_>, tys: &mut FxHash
     }
     let def_id = pat.hir_id.owner.to_def_id();
     if cx.tcx.has_typeck_tables(def_id) {
-        is_mutable_ty(cx, &cx.tcx.typeck_tables_of(def_id).pat_ty(pat), pat.span, tys)
+        is_mutable_ty(
+            cx,
+            &cx.tcx.typeck_tables_of(def_id.expect_local()).pat_ty(pat),
```
Comment (phansch) [inline clippy_lints/src/functions.rs:494]: We have the LocalDefId above: arg.hir_id.owner, no need to convert it back here. Same for all the changes in this file

File: clippy_lints/src/utils/mod.rs [hunk 2]
```diff
@@ -293,7 +293,9 @@ pub fn qpath_res(cx: &LateContext<'_, '_>, qpath: &hir::QPath<'_>, id: hir::HirI
         hir::QPath::Resolved(_, path) => path.res,
         hir::QPath::TypeRelative(..) => {
             if cx.tcx.has_typeck_tables(id.owner.to_def_id()) {
-                cx.tcx.typeck_tables_of(id.owner.to_def_id()).qpath_res(qpath, id)
+                cx.tcx
+                    .typeck_tables_of(id.owner.to_def_id().expect_local())
```
Comment (phansch) [inline clippy_lints/src/utils/mod.rs:297]: I think this is the same:

```suggestion
                    .typeck_tables_of(id.owner)
```

File: clippy_lints/src/utils/usage.rs [hunk 3]
```diff
@@ -19,7 +19,7 @@ pub fn mutated_variables<'a, 'tcx>(expr: &'tcx Expr<'_>, cx: &'a LateContext<'a,
     };
     let def_id = expr.hir_id.owner.to_def_id();
     cx.tcx.infer_ctxt().enter(|infcx| {
-        ExprUseVisitor::new(&mut delegate, &infcx, def_id, cx.param_env, cx.tables).walk_expr(expr);
+        ExprUseVisitor::new(&mut delegate, &infcx, def_id.expect_local(), cx.param_env, cx.tables).walk_expr(expr);
```
Comment (phansch) [inline clippy_lints/src/utils/usage.rs:22]:
```suggestion
        ExprUseVisitor::new(&mut delegate, &infcx, expr.hir_id.owner, cx.param_env, cx.tables).walk_expr(expr);
```

**Comment by matthiaskrgr** [comment]
https://www.githubstatus.com/incidents/cq3jvp8xp8dw

**Pull request merged**